File: ios/Fakarni/FakarniLaunchBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(FakarniLaunchBridge, NSObject)

RCT_EXTERN_METHOD(
  consumePendingLaunch:(RCTPromiseResolveBlock)resolve
  rejecter:(RCTPromiseRejectBlock)reject
)

@end

File: ios/Fakarni/FakarniLaunchBridge.swift
import Foundation
import React

@objc(FakarniLaunchBridge)
final class FakarniLaunchBridge: NSObject {
  @objc(consumePendingLaunch:rejecter:)
  func consumePendingLaunch(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    resolve(FakarniPendingExternalLaunchStore.consume()?.bridgePayload)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    false
  }
}
